### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by aymann121**
Edited Log_utils file to create global graph cvs files and adjacency matricies

Edited plot_utils file to create 3 visualizations based on log_util output files, created a heatmap of node connections, a video of connection progressions throughout rounds and a weighted graphs depicting how often each edge is utilized.

image
image
image

### CODE REVIEW

File: src/algos/base_class.py [hunk 1]
```diff
@@ -339,6 +339,13 @@ def log_metrics(self, stats: Dict[str, Any], iteration: int) -> None:
                 imgs=stats["images"], key="sample_images", iteration=iteration
             )
 
+        #config["num_users"]
```
Comment (tremblerz) [inline src/algos/base_class.py:342]: So is the logic to log a graph at every round or only do it in a posthoc manner (in the plot utils) where we log the graph in the end? I am seeing code in both log_utils and plot_utils so not sure

Comment (aymann121) [reply src/algos/base_class.py:342]: the logic is the log the global csv files for each round of each node. Once the final node of a round finishes, the plotter will work to plot the results of that round based on the data produced by the logger.

Comment (tremblerz) [reply src/algos/base_class.py:342]: But csv files were already getting logged about who is collaborating with whom at every round, why store separate csv files?

Comment (tremblerz) [reply src/algos/base_class.py:342]: In other words, there isn't any information that needs to be captured in addition to what we were already capturing for generating the three network visualizations (post training), correct?

**Comment by tremblerz** [comment]
Based on today's discussion, two remaining tasks -
- Fix the double arrow edges to separate edges.
- All the code should be in `post_hoc` plot_utils

**Pull request closed without merging**